From e991adb91c844d500d18231311ea5d0296a39b5f Mon Sep 17 00:00:00 2001
Message-Id: <e991adb91c844d500d18231311ea5d0296a39b5f.1425610844.git.chang-joon.lee@intel.com>
In-Reply-To: <6fa2ab23adf6d7de781e906d04a0224410262d9b.1425610844.git.chang-joon.lee@intel.com>
References: <6fa2ab23adf6d7de781e906d04a0224410262d9b.1425610844.git.chang-joon.lee@intel.com>
From: Durgadoss R <durgadoss.r@intel.com>
Date: Mon, 9 Feb 2015 16:40:35 +0530
Subject: [PATCH 07/27] MUST_REBASE [VPG]: drivers/video/adf: Disable DRRS
 when PSR is enabled

This patch disables (by not initializing) DRRS if PSR is enabled.
Only one of them is enabled at a time.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed.

Issue: GMINL-5760
Change-Id: I376ef38bd8459851d98acd07bc0ca33fdaeda727
Signed-off-by: Durgadoss R <durgadoss.r@intel.com>
---
 drivers/video/adf/intel/core/common/dp/edp_drrs.c |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/video/adf/intel/core/common/dp/edp_drrs.c b/drivers/video/adf/intel/core/common/dp/edp_drrs.c
index 38c6b19..fa67ac1 100644
--- a/drivers/video/adf/intel/core/common/dp/edp_drrs.c
+++ b/drivers/video/adf/intel/core/common/dp/edp_drrs.c
@@ -95,6 +95,11 @@ int intel_edp_drrs_init(struct intel_pipeline *pipeline)
 	if (!i_pipe->ops->get_preferred_mode)
 		return ret;
 
+	if (dp_pipe->psr.setup_done) {
+		pr_err("PSR enabled, so not enabling DRRS\n");
+		return ret;
+	}
+
 	i_pipe->ops->get_preferred_mode(i_pipe, &preferred);
 	if (!preferred) {
 		pr_err("Failed to obtain edp preferred mode\n");
-- 
1.7.9.5

